src/odemis/gui/cont/tabs.py

```diff
@@ -4455,7 +4469,496 @@ def _on_align_move_done(self, f=None):
+                return
+            except Exception as ex:
+                logging.error("Aligner move failed: %s", ex)
+                return
+
+        future_stage_move.add_done_callback(move_aligner)
+
+    def _click_align_mode_button(self, clicked_align_mode):
+        """
+        Setter for the current alignment mode. Changes the mode with the corresponding streams to be displayed, the
+        instructions and the controls. Also takes care of the toggling of the buttons.
+
+        :param clicked_align_mode (GraphicRadioButton): Button pressed by the user.
+        """
+        clicked_align_mode = clicked_align_mode.GetEventObject()
+        clicked_align_mode.SetToggle(True)
+        # Untoggle all the other buttons.
+        for btn in self._align_btn_functions:
+            if btn is not clicked_align_mode:
+                btn.SetToggle(False)
+
+        self._align_btn_functions[clicked_align_mode]()  # Calls the setter of the correct mode
+        return clicked_align_mode
+
+    def _set_z_alignment_mode(self):
+        """
+        Sets the Z alignment mode with the correct streams, stream controllers, instructions and alignment controls.
+        """
+        self.stream_bar_controller.pauseAllStreams()
+        self._set_top_and_bottom_stream_and_settings(self._FIB_view_and_control,
+                                                     self._SEM_view_and_control)
+
+        # Adjust the overlay to a horizontal line in the FIB viewport
+        for viewport in self.panel.pnl_two_streams_grid.viewports:
+            if viewport.view.stream_classes is acqstream.FIBStream:
+                for FIB_line_overlay in viewport.canvas.view_overlays:
+                    if isinstance(FIB_line_overlay, CenteredLineOverlay):
+                        FIB_line_overlay.shape = HORIZONTAL_LINE
+                        break
+
+        doc_path = pkg_resources.resource_filename("odemis.gui", "doc/enzel_z_alignment.html")
+        self.panel.html_alignment_doc.LoadPage(doc_path)
+
+        self.panel.controls_step_size_slider.SetRange(100e-9, 10e-6)
+        self.panel.controls_step_size_slider.set_position_value(100e-9)
+
+        self.panel.pnl_z_align_controls.Show(True)
+        self.panel.pnl_sem_align_controls.Show(False)
+        self.panel.pnl_flm_align_controls.Show(False)
+        fix_static_text_clipping(self.panel)
+        self.tab_data_model.align_mode.value = "Z alignment"
+
+    def _z_alignment_controls(self, axis, proportion, evt):
+        """
+        Links the Z alignment mode control buttons to the stage and updates the correct metadata.
+
+        :param axis (str): axis which is controlled by the button.
+        :param proportion (int): direction of the movement represented by the button (-1/1)
+        :param evt (GenButtonEvent): Clicked event
+        """
+        # Only proceed if there is no currently running target_position
+        if hasattr(self, "_z_move_future") and self._z_move_future._state == RUNNING:
```

Comment:
To check the state of a future, use `not self._z_move_future.done()` .
You sould also set _z_move_future at init as None, so that you can more nicely write `if self.z_move_future ...`